lms/djangoapps/grades/new/course_grade.py

```diff
@@ -24,8 +27,39 @@ class CourseGrade(object):
     def __init__(self, student, course, course_structure):
         self.student = student
         self.course = course
+        self.course_version = getattr(course, 'course_version', None)
+        self.course_edited_timestamp = getattr(course, 'subtree_edited_on', None)
         self.course_structure = course_structure
-        self.chapter_grades = []
+        self._percent = None
+        self._letter_grade = None
+        self.subsection_grade_factory = SubsectionGradeFactory(self.student, self.course, self.course_structure)
+
+    @classmethod
+    def load_persisted_grade(cls, user, course, course_structure, current_grading_policy_hash):
```

Comment:
I think the names in `SubsectionGrade` are still pretty faithful to the purpose of the method they describe; `init_from_model` and `init_from_structure` are both fairly simple initializers with no real logic, whereas this method does some work of its own. I wouldn't oppose changing `init` to `load` in those method names though; I don't really see a downside.